direct/tpm2/policy_test.go

```diff
@@ -281,6 +282,189 @@ func TestPolicyOrUpdate(t *testing.T) {
 	}
 }
 
+func TestPolicyPCRUpdate(t *testing.T) {
+	thetpm, err := simulator.OpenSimulator()
+	if err != nil {
+		t.Fatalf("could not connect to TPM simulator: %v", err)
+	}
+	defer thetpm.Close()
+
+	// Use a trial session to calculate this policy
+	sess, cleanup2, err := PolicySession(thetpm, tpm.AlgSHA1, 16, Trial())
+	if err != nil {
+		t.Fatalf("setting up policy session: %v", err)
+	}
+	defer func() {
+		t.Helper()
+		if err := cleanup2(); err != nil {
+			t.Errorf("cleaning up policy session: %v", err)
+		}
+	}()
+
+	PCRs, err := CreatePCRSelection([]int{0, 1, 2, 3, 7})
+	if err != nil {
+		t.Fatalf("Failed to create PCRSelection")
+	}
+	selection := tpml.PCRSelection{
+		PCRSelections: []tpms.PCRSelection{
+			{
+				Hash:      tpm.AlgSHA1,
+				PCRSelect: PCRs,
+			},
+		},
+	}
+
+	pcrRead := PCRRead{
+		PCRSelectionIn: selection,
+	}
+
+	pcrReadRsp, err := pcrRead.Execute(thetpm)
+	if err != nil {
+		t.Fatalf("Failed to read PCRSelection")
+	}
+
+	var expectedVal []byte
+	for _, digest := range pcrReadRsp.PCRValues.Digests {
+		expectedVal = append(expectedVal, digest.Buffer...)
+	}
+	t.Logf("expectedVal=%x", expectedVal)
+
+	// Hash algorithm must match the one in PolicySession.
+	expectedDigest := sha1.Sum(expectedVal)
```

Comment:
This is heavily duplicated in the two tests. Could we pull it out into a helper?

```
func getExpectedPCRDigest(t *testing.T, tpm transport.TPM, pcrs []int, alg tpm.HashAlg) []byte {
```

inside you can also convert the tpm.HashAlg to a crypto.Hash to get the digest

Reply:
I will pull out the process needed to produce an expected digest, but I think everything before that is essential to code readability and testing down the line. Ex. selection is used in the table tests.

Reply:
Resolved. Pulled out process of reading PCR selection and processing to get PCRDigest.